tests/countries/test_sweden.py

```diff
@@ -173,6 +175,79 @@ def test_christmas(self):
     def test_sundays(self):
         self.assertSundays(Sweden)  # Sundays are considered holidays in Sweden.
 
+    def test_de_facto_holidays(self):
+        """Test DE_FACTO category holidays."""
+        name_midsummer = "Midsommarafton"
+        name_christmas = "Julafton"
+        name_newyear = "Nyårsafton"
+
+        # These should NOT be in default (PUBLIC) category
+        self.assertNoHolidayName(name_midsummer)
+        self.assertNoHolidayName(name_christmas)
+        self.assertNoHolidayName(name_newyear)
+
+        # These should NOT be in BANK category
+        self.assertNoBankHolidayName(name_midsummer)
+        self.assertNoBankHolidayName(name_christmas)
+        self.assertNoBankHolidayName(name_newyear)
+
+        # These should NOT be in OPTIONAL category
+        self.assertNoOptionalHolidayName(name_midsummer)
+        self.assertNoOptionalHolidayName(name_christmas)
+        self.assertNoOptionalHolidayName(name_newyear)
+
+        # These SHOULD be in DE_FACTO category
+        # Midsummer Eve dates
+        dts_midsummer = (
+            "2020-06-19",
+            "2021-06-25",
+            "2022-06-24",
+            "2023-06-23",
+            "2024-06-21",
+            "2025-06-20",
+        )
+        self.assertDeFactoHolidayName(name_midsummer, dts_midsummer)
+
+        # Christmas Eve - every year
+        self.assertDeFactoHolidayName(
+            name_christmas, (f"{year}-12-24" for year in self.full_range)
+        )
+
+        # New Year's Eve - every year
+        self.assertDeFactoHolidayName(name_newyear, (f"{year}-12-31" for year in self.full_range))
```

Comment:
IMO this is currently a duplication of the existing `test_midsummer_eve`, `test_christmas_eve` and `test_new_years_eve` - we can safely remove this
```suggestion
```